Library/Homebrew/utils/user.rb

```diff
@@ -0,0 +1,12 @@
+require "system_command"
+require "etc"
+
+USER = Etc.getlogin
```

Comment:
Why all in caps?

Reply:
Because it's a constant and not a class.

Reply:
What is returned, a string? Feels like a class which encapsulates that and has the `gui?` method on it might be more idomatic.

Reply:
A class would need to forward every class method to `Etc.getlogin`, which would be much more complex than simply adding a method to this string.

Reply:
> would be much more complex than simply adding a method to this string.

Are you thinking about code complexity, or rather complexity at runtime?

Would you consider the following too complex?

```
require 'delegate'
require 'etc'

class User < DelegateClass(String)
  def gui?
    # …
  end
end

USER = User.new(Etc.getpwuid(Process.euid).name)
```